# Hai there
This reposotory is a collection of all the bioinformatics related coding.<br>

<div style="text-align: center;">
  <h1>Identification of Biomarkers for Personalized Medicine in Psoriatic Arthritis</h1>
  <p>By</p>
  <h2>David Franklin D.</h2>
  <p>A Project work submitted for the certification course Bio-informatics</p>
  <h3>Bversity</h3>
  <p>2024</p>
</div>

## Introduction
<p>Psoriatic arthritis (PsA) is a chronic autoimmune disease characterized by
inflammation of the skin and joints. It affects approximately 30% of individuals with
psoriasis, a common skin condition characterized by red, scaly patches. Psoriatic
arthritis not only causes joint pain, stiffness, and swelling but can also lead to
progressive joint damage and disability if left untreated.</p>

<img src="Bversity_project/attachments/PsA_hands.png" alt="Description" width="200" height="200"> <img src="Bversity_project/attachments/jointsPsA.png" alt="Description" width="200" height="200">



## Objectives:
<img src="Bversity_project/attachments/PsA_patient_observation.png" alt="Description" width="500" height="350">
<p>The primary objective of this project is to leverage publicly available gene
expression data to identify potential drug targets for personalized medicine in
Psoriatic Arthritis (PsA). Specifically, the project aims to:</p>
<p>1. Utilize NCBI GEO Data: Access and analyze gene expression data obtained
from the National Center for Biotechnology Information (NCBI) Gene Expression
Omnibus (GEO) database. The dataset includes gene expression profiles from skin
samples of PsA patients and normal controls, providing valuable insights into the
molecular mechanisms underlying the disease.</p>
<p>2. Identify Differentially Expressed Genes (DEGs): Employ advanced
bioinformatics techniques to identify genes that are differentially expressed between
lesional skin, non-lesional skin, and normal skin samples. By comparing gene
expression patterns across different disease states and control samples, the project
aims to pinpoint key molecular alterations associated with PsA pathogenesis.</p>
<p>3. Conduct Pathway Analysis and Functional Annotation: Perform pathway
analysis and functional annotation of the identified DEGs to elucidate the underlying
biological processes and pathways dysregulated in Psoriatic Arthritis. By integrating
gene expression data with biological pathway databases, the project seeks to identify
enriched pathways and molecular functions relevant to disease progression.</p>
<p>4. Prioritize Potential Drug Targets: Prioritize potential drug targets based on
their biological significance and relevance to PsA pathophysiology. Through
comprehensive analysis of gene expression data and pathway information, the project
aims to identify promising candidates for therapeutic intervention in Psoriatic
Arthritis.</p>

## Complete Analysis

[Full Report]((David_Franklin)_Batch#1_Bioinformatics_Placement_(Project_Report).pdf)

[Powerpoint presentation](Bversity_project/(David_Franklin)_Batch#1_Bioinformatics_Placement_Program_Presentation.pptx)

Analysis Steps and code:-
1. [1_Pre_processing](Bversity_project/1_Pre_processing.md)
2. [2_visualizatio](Bversity_project/2_visualization.md)
3. [3_functional_analysis](Bversity_project/3_functional_analysis_&_.md)
4. [4_Network_analysis](Bversity_project/4_Network_analysis.md)

## Tools used 
R, NCBI-GEO database, cytoscape
---